### PR TITLE
refs #10948 - accounting for unlimited_content_hosts=false, BZ1214675

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

**Comment by komidore64** [comment]
must write test!

File: app/controllers/katello/api/v2/host_collections_controller.rb [hunk 1]
```diff
@@ -151,6 +152,12 @@ def copy
 
     private
 
+    def find_max_content_host_count
+      if !params['unlimited_content_hosts'] && params['max_content_hosts'].nil?
+        fail HttpErrors::BadRequest, _("unlimited_content_hosts is false but there is no value for max_content_hosts")
+      end
+    end
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/host_collections_controller.rb]: would this make sense as a model validation?

Comment (komidore64) [inline app/controllers/katello/api/v2/host_collections_controller.rb]: I realized after creating this PR that this logic wasn't quite correct. I saw that I broke a test. Better logic incoming!

**Comment by komidore64** [comment]
test written

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**